services/movie_session.py

```diff
@@ -0,0 +1,56 @@
+import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession, CinemaHall, Movie
+
+
+def create_movie_session(
+    movie_show_time: datetime, movie_id: int, cinema_hall_id: int
+) -> None:
+    cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+    movie = Movie.objects.get(id=movie_id)
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall=cinema_hall,
+        movie=movie
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet:
+    movie_session_query = MovieSession.objects.all()
+
+    if session_date is not None:
+        movie_session_query = movie_session_query.filter(
+            show_time__date=session_date
+        )
+
+    return movie_session_query
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+    session_id: int,
+    show_time: str = None,
+    movie_id: int = None,
+    cinema_hall_id: int = None,
+) -> None:
+    movie_session = MovieSession.objects.all()
```

Comment:
Why are you trying to get all sessions and then filtering them by 'session_id' to have a single one? It is simpler to use MovieSession.objects.get(id=session_id) - you will have an object instantly instead of filtering a QuerySet.

Reply:
no need to use all here, u can use filter as below